packages/xo-server/src/index.js

```diff
@@ -282,11 +281,10 @@ async function setUpPassport(express, xo, { authentication: authCfg }) {
     new LocalStrategy(
       { passReqToCallback: true },
       async (req, username, password, done) => {
-        const zone = Zone.current.fork('webAuthentication')
-        zone.data.userIp = req.socket.remoteAddress
         try {
-          const { user } = await zone.run(() =>
-            xo.authenticateUser({ username, password })
+          const { user } = await xo.authenticateUser(
+            { username, password },
+            { ip: req.socket.remoteAddress }
```

Comment:
Express provides [`req.ip`](https://devdocs.io/express/index#req.ip) which is better because it works even when behind a reverse proxy, please check whether we can use this.